pennylane/gradients/general_shift_rules.py

```diff
@@ -397,9 +397,9 @@ def _copy_and_shift_params(tape, indices, shifts, multipliers, cast=False):
         # Shift copied parameter
         new_params = list(op.data)

```

Comment:
We're needing to cast `new_params` to float64 too.

Reply:
If I cast it with say
```
new_params[p_idx] = qml.math.cast(new_params[p_idx], np.float64)
```
then I get an error 
```
ImportError: autoray couldn't find function 'astype' for backend 'builtins'.
```
Do you know what to do with that?

Reply:
Think I found a patch by branching `cast`.

Reply:
My solution was similarly patchy. It breaks some other tests. Playing with yours, I have issues with autograd and complex in other tests.

Reply:
How hard can it be just to convert data to float!

Reply:
Too hard! I instead added a condition to skip casting for integral types.